docs/src/main/asciidoc/rest-client.adoc

```diff
@@ -432,6 +432,113 @@ You should see a JSON object containing some basic information about Greece.
+annotation to specify a `io.quarkus.test.common.QuarkusTestResourceLifecycleManager` which can start the service and supply configuration
+values that Quarkus will use.
+
+[NOTE]
+====
+For more details about `@QuarkusTestResource` refer to  link:getting-started-testing#quarkus-test-resource[this] part of the documentation.
+====
+
+Let's create an implementation of `QuarkusTestResourceLifecycleManager` called `WiremockCountries` like so:
+
+[source,java]
+----
+package org.acme.rest.client;
+
+import java.util.Collections;
+import java.util.Map;
+
+import com.github.tomakehurst.wiremock.WireMockServer;
+import io.quarkus.test.common.QuarkusTestResourceLifecycleManager;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.*; // <1>
+
+public class WiremockCountries implements QuarkusTestResourceLifecycleManager {  // <2>
+
+	private WireMockServer wireMockServer;
+
+	@Override
+	public Map<String, String> start() {
+		wireMockServer = new WireMockServer();
+		wireMockServer.start(); // <3>
+
+		stubFor(get(urlEqualTo("/v2/name/GR"))   // <4>
+				.willReturn(aResponse()
+						.withHeader("Content-Type", "application/json")
+						.withBody( // <5>
+						"""
+						[
+							{
+								"name": "Ελλάδα",
+								"capital": "Αθήνα"
+							}
+						]
+						"""
+						)));
+
+		stubFor(get(urlMatching(".*")).atPriority(10).willReturn(aResponse().proxiedFrom("https://restcountries.eu/rest")));   // <6>
+
+		return Collections.singletonMap("org.acme.getting.started.country.CountriesService/mp-rest/url", wireMockServer.baseUrl()); // <7>
+	}
+
+	@Override
+	public void stop() {
+		if (null != wireMockServer) {
+			wireMockServer.stop();  // <8>
+		}
+	}
+}
+----
+
+<1> Statically importing the methods in the Wiremock package makes it easier to read the test.
+<2> The `start` method is invoked by Quarkus before any test is run and returns a `Map` of configuration properties that apply during the test execution.
+<3> Launch Wiremock.
+<4> Configure Wiremock to stub the calls to `/v2/name/GR` by returning a specific canned response.
+<5> This examples uses Text Blocks from Java 14 because it makes the code a lot easier to read.
+<6> All HTTP calls that have not been stubbed are handled by calling the real service. This is done for demonstration purposes, as it is not something that would usually happen in a real test.
+<7> As the `start` method returns configuration that applies for tests, we set the rest-client property that controls the base URL which is used by the implementation
+of `CountriesService` to the base URL where Wiremock is listening for incoming requests.
+<8> When all tests have finished, shutdown Wiremock.
+
+
+The `CountriesResourceTest` test class needs to be annotated like so:
+
+[source,java]
+----
+@QuarkusTest
+@QuarkusTestResource(WiremockCountries.class)
+public class CountriesResourceTest {
+
+}
+----
+
+[WARNING]
+====
+`@QuarkusTestResource` applies to all tests, not just `CountriesResourceTest`.
```

Comment:
I wonder how well this will scale given you will probably want different mock implementations for different tests.

Works well for one test, but when you have several of them?

Reply:
We don't yet have the capability to add different `@QuarkusTestResourc` to different tests, but we will very soon.

Reply:
OK, that's important information :).

If we do, then I concur it makes sense to document it.

Reply:
Yeah, it's just a matter of me or Stuart looking at it. It should be easy, I just haven't got around to doing it